ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -890,3 +929,33 @@ mergeListsBy cmp all_lists = merge_lists all_lists
         []     -> []
         [xs]   -> xs
         lists' -> merge_lists lists'
+
+
+getCompletionPrefix :: (Monad m) => Position -> VFS.VirtualFile -> m (Maybe PosPrefixInfo)
+getCompletionPrefix pos@(Position l c) (VFS.VirtualFile _ _ ropetext) =
+      return $ Just $ fromMaybe (PosPrefixInfo "" "" "" pos) $ do -- Maybe monad
```

Comment:
This is quite weird: you're unconditionally returning a `Just`, but you _have_ a `Maybe` already, but you're swapping out the `Nothing` case with a default value? Quite confusing. Maybe you just want to also drop the `Maybe` from the return type?

Reply:
I see, this was here before. Well, it makes no sense so we might as well improve it.